inc/impact.class.php

```diff
@@ -895,6 +895,22 @@ public static function searchAsset(
          );
       }
 
+      if ($item->mayBeDeleted()) {
+         $criteria["$table.is_deleted"] = 0;
+         $base_request = array_merge(
+            $base_request,
+            $criteria
+         );
```

Comment:
```suggestion
         $base_request['WHERE']["$table.is_deleted"] = 0;
```

Reply:
At the beginning, i do this, but without array_merge, the SQL do not use parameters
`SELECT `glpi_computers`.`id`, `glpi_computers`.`name` FROM `glpi_computers` WHERE (`glpi_computers`.`entities_id` IN ('1')) LIMIT 20`

Reply:
I'm not sure, but anyways: you use  `$criteria` that does not exists, and you do not reset it. Second call will contains first one; this is to avoid.
You probably should handle that before entities restrictions are seted up.

Please try with @cedric-anne proposal at the good place and if that does not work, we need to see what the array looks like (and also, please quote correctly SQL, this is really difficult to read - I have to edit your comment).